DataManagementSystem/Client/ConsistencyChecks.py

```diff
@@ -0,0 +1,836 @@
+      self.__reportProblematicReplicas( missingReplicas, se, 'PFNMissing' )
+    lostReplicas = []
+    unavailableReplicas = []
+    zeroSizeReplicas = []
+    # If the files are not accessible
+    for lfn, lfnMetadata in lfnMetadataDict.items():
+      if lfnMetadata['Lost']:
+        lostReplicas.append( ( lfn, 'deprecatedUrl', se, 'PFNLost' ) )
+      if lfnMetadata['Unavailable']:
+        unavailableReplicas.append( ( lfn, 'deprecatedUrl', se, 'PFNUnavailable' ) )
+      if lfnMetadata['Size'] == 0:
+        zeroSizeReplicas.append( ( lfn, 'deprecatedUrl', se, 'PFNZeroSize' ) )
+    if lostReplicas:
+      self.__reportProblematicReplicas( lostReplicas, se, 'PFNLost' )
+    if unavailableReplicas:
+      self.__reportProblematicReplicas( unavailableReplicas, se, 'PFNUnavailable' )
+    if zeroSizeReplicas:
+      self.__reportProblematicReplicas( zeroSizeReplicas, se, 'PFNZeroSize' )
+    gLogger.info( 'Checking the integrity of physical files at %s complete' % se )
+    return S_OK( lfnMetadataDict )
+
+  ##########################################################################
+  #
+  # This section contains the specific methods for SE->LFC checks
+  #
+
+  def storageDirectoryToCatalog( self, lfnDir, storageElement ):
+    """ This obtains the file found on the storage element in the supplied directories and determines whether they exist in the catalog and checks their metadata elements
+    """
+    gLogger.info( "-" * 40 )
+    gLogger.info( "Performing the SE->LFC check at %s" % storageElement )
+    gLogger.info( "-" * 40 )
+    if type( lfnDir ) in types.StringTypes:
+      lfnDir = [lfnDir]
+    res = self.__getStorageDirectoryContents( lfnDir, storageElement )
+    if not res['OK']:
+      return res
+    storageFileMetadata = res['Value']
+    if storageFileMetadata:
+      return self.__checkCatalogForSEFiles( storageFileMetadata, storageElement )
+    return S_OK( {'CatalogMetadata':{}, 'StorageMetadata':{}} )
+
+  def __checkCatalogForSEFiles( self, storageMetadata, storageElement ):
+    gLogger.info( 'Checking %s storage files exist in the catalog' % len( storageMetadata ) )
+
+
+    res = self.fc.getReplicas( storageMetadata )
+    if not res['OK']:
+      gLogger.error( "Failed to get replicas for LFN", res['Message'] )
+      return res
+    failedLfns = res['Value']['Failed']
+    successfulLfns = res['Value']['Successful']
+    notRegisteredLfns = []
+    
+    for lfn in storageMetadata:
+      if lfn in failedLfns: 
+        if 'No such file or directory' in failedLfns[lfn]:
+          notRegisteredLfns.append( ( lfn, 'deprecatedUrl', storageElement, 'LFNNotRegistered' ) )
+          failedLfns.pop( lfn )
+      elif storageElement not in successfulLfns[lfn]:
+        notRegisteredLfns.append( ( lfn, 'deprecatedUrl', storageElement, 'LFNNotRegistered' ) )
+
+    if notRegisteredLfns:
+      self.__reportProblematicReplicas( notRegisteredLfns, storageElement, 'LFNNotRegistered' )
+    if failedLfns:
```

Comment:
Is it correct that you stop if you couldn't get the replicas even for one file ?